src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -887,6 +895,61 @@ bool ClusterTaskManager::ReturnCpuResourcesToBlockedWorker(
 void ClusterTaskManager::ScheduleAndDispatchTasks() {
   SchedulePendingTasks();
   DispatchScheduledTasksToWorkers(worker_pool_, leased_workers_);
+  // TODO(swang): Spill from waiting queue first? Otherwise, we may end up
```

Comment:
This sounds like a good idea? Why do you not do that here? Any edge case in your mind?

Reply:
It's a bit hard to reason about. I'd rather do this if there is actually a bug associated with it.